app/upgrades/argon/upgrade.go

```diff
@@ -1,50 +1,45 @@
 package argon
 
 import (
-	"fmt"
-
 	"cosmossdk.io/math"
 	cctpkeeper "github.com/circlefin/noble-cctp/x/cctp/keeper"
 	cctptypes "github.com/circlefin/noble-cctp/x/cctp/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
-	upgradeTypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
 	fiattokenfactorykeeper "github.com/strangelove-ventures/noble/x/fiattokenfactory/keeper"
-	paramauthoritykeeper "github.com/strangelove-ventures/paramauthority/x/params/keeper"
 )
 
 func CreateUpgradeHandler(
 	mm *module.Manager,
 	configurator module.Configurator,
-	fiatTFKeeper *fiattokenfactorykeeper.Keeper,
-	paramauthoritykeeper paramauthoritykeeper.Keeper,
 	cctpKeeper *cctpkeeper.Keeper,
-) upgradeTypes.UpgradeHandler {
-	return func(ctx sdk.Context, _ upgradeTypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
-		var cctpAuthority string
-		paramAuthority := paramauthoritykeeper.GetAuthority(ctx)
-		if ctx.ChainID() == TestnetChainID {
-			cctpAuthority = paramAuthority
-		} else {
-			owner, ok := fiatTFKeeper.GetOwner(ctx)
-			if !ok {
-				return nil, fmt.Errorf("fiat token factory owner not found")
-			}
+	fiatTokenFactoryKeeper *fiattokenfactorykeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		if ctx.ChainID() == MainnetChainID {
```

Comment:
we don't need this check anymore since this handler will only run on mainnet based on the versions used, but it also doesn't hurt.